clippy_lints/src/excessive_nesting.rs

```diff
@@ -0,0 +1,329 @@
+                self.visit_block(block);
+
+                if let Some(expr) = else_expr {
+                    self.visit_expr(expr);
+                }
+            },
+            ExprKind::While(expr, block, ..) => {
+                self.visit_expr(expr);
+                self.visit_block(block);
+            },
+            ExprKind::ForLoop(pat, expr, block, ..) => {
+                self.visit_pat(pat);
+                self.visit_expr(expr);
+                self.visit_block(block);
+            },
+            ExprKind::Loop(block, ..)
+            | ExprKind::Block(block, ..)
+            | ExprKind::Async(.., block)
+            | ExprKind::TryBlock(block) => {
+                self.visit_block(block);
+            },
+            ExprKind::Match(expr, arms) => {
+                self.visit_expr(expr);
+
+                for arm in arms {
+                    self.visit_arm(arm);
+                }
+            },
+            ExprKind::Closure(closure) => self.visit_expr(&closure.body),
+            ExprKind::Range(start, end, ..) => {
+                if let Some(expr) = start {
+                    self.visit_expr(expr);
+                }
+                if let Some(expr) = end {
+                    self.visit_expr(expr);
+                }
+            },
+            ExprKind::Break(.., expr) | ExprKind::Ret(expr) | ExprKind::Yield(expr) | ExprKind::Yeet(expr) => {
+                if let Some(expr) = expr {
+                    self.visit_expr(expr);
+                }
+            },
+            ExprKind::Struct(struct_expr) => {
+                for field in &struct_expr.fields {
+                    self.visit_expr(&field.expr);
+                }
+            },
+            _ => (),
+        }
+    }
+
+    fn visit_fn(&mut self, fk: FnKind<'_>, _: Span, _: NodeId) {
+        match fk {
+            FnKind::Fn(.., block) if let Some(block) = block => self.visit_block(block),
+            FnKind::Closure(.., expr) => self.visit_expr(expr),
+            // :/
+            FnKind::Fn(..) => (),
+        }
+    }
+
+    fn visit_item(&mut self, item: &Item) {
+        match &item.kind {
+            ItemKind::Static(static_item) if let Some(expr) = static_item.expr.as_ref() => self.visit_expr(expr),
+            ItemKind::Const(const_item) if let Some(expr) = const_item.expr.as_ref() => self.visit_expr(expr),
+            ItemKind::Fn(fk) if let Some(block) = fk.body.as_ref() => self.visit_block(block),
+            ItemKind::Mod(.., mod_kind)
+                if let ModKind::Loaded(items, Inline::Yes, ModSpans { inner_span, ..}) = mod_kind =>
+            {
+                self.nest_level += 1;
+
+                check_indent(self, *inner_span);
+
+                self.nest_level -= 1;
+            }
+            ItemKind::Trait(trit) => check_trait_and_impl(self, item, &trit.items),
+            ItemKind::Impl(imp) => check_trait_and_impl(self, item, &imp.items),
+            _ => (),
```

Comment:
We can use `walk_item` here & merge the `impl`/`trait`/`mod foo { .. }` cases

```suggestion
            ItemKind::Trait(_) | ItemKind::Impl(_) | ItemKind::Mod(.., ModKind::Loaded(_, Inline::Yes, _)) => {
                self.nest_level += 1;

                if !check_indent(self, item.span) {
                    walk_item(self, item);
                }

                self.nest_level -= 1;
            },
            // Don't visit non-inline modules
            ItemKind::Mod(..) => {},
            _ => walk_item(self, item),
```